x-pack/plugins/stack_connectors/server/connector_types/bedrock/bedrock.ts

```diff
@@ -125,33 +143,105 @@ export class BedrockConnector extends SubActionConnector<Config, Secrets> {
   }
 
   /**
-   * takes in an array of messages and a model as input, and returns a promise that resolves to a string.
-   * The method combines the messages into a single prompt formatted for bedrock,sends a request to the
-   * runApi method with the prompt and model, and returns the trimmed completion from the response.
-   * @param messages An array of message objects, where each object has a role (string) and content (string) property.
+   *  NOT INTENDED TO BE CALLED DIRECTLY
+   *  call invokeStream instead
+   *  responsible for making a POST request to a specified URL with a given request body.
+   *  The response is then processed based on whether it is a streaming response or a regular response.
+   * @param body The stringified request body to be sent in the POST request.
    * @param model Optional model to be used for the API request. If not provided, the default model from the connector will be used.
    */
+  private async streamApi({
+    body,
+    model: reqModel,
+  }: StreamActionParams): Promise<StreamingResponse> {
+    // set model on per request basis
+    const model = reqModel ? reqModel : this.model;
+
+    const signed = this.signRequest(body, `/model/${model}/invoke-with-response-stream`, true);
+
+    const response = await this.request({
+      ...signed,
+      url: `${this.url}/model/${model}/invoke-with-response-stream`,
```

Comment:
NIT: store all `path` instead of just `model`, so we don't have to duplicate it.

```suggestion
    const path = `/model/${reqModel ?? this.model}/invoke-with-response-stream`;
    const signed = this.signRequest(body, path, true);
    const response = await this.request({
      ...signed,
      url: `${this.url}${path}`,
```
Same at [runApi](https://github.com/elastic/kibana/pull/170815/files#diff-f0215878b5d4d091a224b60506563a1872aa30f9df48d8a8a000b40b331ca758R132)